cumulusci/tasks/robotframework/debugger/ui.py

```diff
@@ -0,0 +1,228 @@
+        """Run the next step in the test
+
+        This will run the line where the test is currently paused, and then
+        stop at the next keyword in the same test case.
+        """
+        self.listener.do_step()
+        return True
+
+    def do_vars(self, arg):
+        """Print the value of all known variables"""
+        vars = self.builtin.get_variables()
+        vars = [["Variable", "Value"]] + [x for x in map(list, sorted(vars.items()))]
+        CliTable(vars).echo()
+
+    def do_where(self, arg):
+        """
+        Print the current robot stack (hierarchy of suites and tests,
+        ending with the current keyword
+        """
+
+        prefix = "  "
+        for i, x in enumerate(self.listener.stack):
+            indent = prefix * i
+            print("{}: {}-> {}".format(i, indent, x.longname), file=self.stdout)
+        print("", file=self.stdout)
+
+    def _highlight_element(self, element, style=None):
+        """Highlight a Selenium Webdriver element
+
+        style can be None, a string with css styles, or a dict of css styles
+        element needs to be an instance of WebElement
+        """
+
+        if style is None:
+            element_style = """
+            box-shadow: 0px 1px 4px 2px inset #FFFF00;
+            """
+        elif isinstance(style, dict):
+            element_style = "\n".join(
+                "{}: {};".format(key, value) for key, value in style.items()
+            )
+        else:
+            element_style = style
+
+        original_style = element.get_attribute("style")
+        new_style = original_style + element_style
+        self.selenium.driver.execute_script(
+            """
+            if (!arguments[0].hasAttribute("data-original-style")) {
+                console.log('adding data-original-style...');
+                /* only save the original style if we haven't already done so */
+                arguments[0].setAttribute('data-original-style', arguments[0].getAttribute('style'));
+            };
+            arguments[0].setAttribute('style', arguments[1]);
+
+        """,
+            element,
+            new_style,
+        )
+
+    def _restore_element_style(self, element):
+        """Restore the element style from the data-original-style attribute
+
+        This is to undo the effects of _highlight_element
+        """
+        js = """
+        if (arguments[0].hasAttribute('data-original-style')) {
+            var original_style = arguments[0].getAttribute('data-original-style');
+            arguments[0].setAttribute('style', original_style);
+            arguments[0].removeAttribute('data-original-style');
+            return true;
+        } else {
+            return false;
+        }
+        """
+
+        result = self.selenium.driver.execute_script(js, element)
+        if result is False:
+            self.builtin.log(
+                "unable to restore style; original style not found", "DEBUG"
+            )
+        return result
```

Comment:
These last 2 methods are missing test coverage.

Reply:
All methods in this file have coverage.